plugins/fluentd_telemetry_plugin/src/merge_configuration_files.py

```diff
@@ -0,0 +1,68 @@
+"""
+@copyright:
+    Copyright (C) Mellanox Technologies Ltd. 2014-2020.  ALL RIGHTS RESERVED.
+
+    This software product is a proprietary product of Mellanox Technologies
+    Ltd. (the "Company") and all right, title, and interest in and to the
+    software product, including all associated intellectual property rights,
+    are and shall remain exclusively with the Company.
+
+    This software product is governed by the End User License Agreement
+    provided with the software product.
+
+@author: Miryam Schwartz
+@date:   Oct 14, 2024
+"""
+import configparser
+import logging
+import sys
+import os
+
+def merge_ini_files(old_file_path, new_file_path):
+    # Check if files exist
+    if not os.path.isfile(old_file_path):
+        logging.error("file %s does not exist.", old_file_path)
+        sys.exit(1)
+    if not os.path.isfile(new_file_path):
+        logging.error("file %s does not exist.", new_file_path)
+        sys.exit(1)
+
+    # Create a configparser object
+    config_old = configparser.ConfigParser()
+    config_new = configparser.ConfigParser()
+
+    # Read the old and new files
+    try:
+        config_old.read(old_file_path)
+        config_new.read(new_file_path)
+
+    except configparser.Error as e:
+        logging.error("Failed to parse configurations files: %s", e)
+        sys.exit(1)
+
+    # Merge configurations
+    for section in config_new.sections():
+        if not config_old.has_section(section):
+            config_old.add_section(section)
+        for option in config_new.options(section):
+            # If option exists in the old file, retain the old value
+            if not config_old.has_option(section, option):
```

Comment:
what if an existing option or section is removed from the new conf file?
we should also remove it from the existing (old) conf file,

Reply:
right. I changed the logic